flink-rpc/flink-rpc-akka/src/test/java/org/apache/flink/runtime/rpc/akka/ContextClassLoadingSettingTest.java

```diff
@@ -107,49 +108,87 @@ void shutdown() throws InterruptedException, ExecutionException, TimeoutExceptio
-        assertIsFlinkClassLoader(contextClassLoader.get());
+        final ClassLoader contextClassLoader =
+                akkaRpcService
+                        .getScheduledExecutor()
+                        .schedule(
+                                () -> Thread.currentThread().getContextClassLoader(),
+                                0,
+                                TimeUnit.MILLISECONDS)
+                        .get();
+        assertIsFlinkClassLoader(contextClassLoader);
     }
 
     @Test
-    void testAkkaRpcService_ExecuteCallableResultCompletedWithFlinkContextClassLoader()
+    void testAkkaRpcService_ScheduleRunnableSetsFlinkContextClassLoader()
             throws ExecutionException, InterruptedException {
-
-        final CompletableFuture<Void> blocker = new CompletableFuture<>();
-
-        final CompletableFuture<ClassLoader> contextClassLoader =
-                runWithContextClassLoader(
+        final CompletableFuture<ClassLoader> contextClassLoader = new CompletableFuture<>();
+        akkaRpcService
+                .getScheduledExecutor()
+                .schedule(
                         () ->
-                                akkaRpcService
-                                        .execute((Callable<Void>) blocker::get)
-                                        .thenApply(
-                                                ignored ->
-                                                        Thread.currentThread()
-                                                                .getContextClassLoader()),
-                        testClassLoader);
-        blocker.complete(null);
+                                contextClassLoader.complete(
+                                        Thread.currentThread().getContextClassLoader()),
+                        5,
+                        TimeUnit.MILLISECONDS);
+        assertThat(contextClassLoader.get()).isSameAs(pretendFlinkClassLoader);
+    }
 
-        assertIsFlinkClassLoader(contextClassLoader.get());
+    @Test
+    void testAkkaRpcService_ScheduleRunnableWithFixedRateSetsFlinkContextClassLoader() {
+        final List<ClassLoader> contextClassLoaders = new ArrayList<>(2);
+        akkaRpcService
+                .getScheduledExecutor()
+                .scheduleAtFixedRate(
+                        () -> {
+                            if (contextClassLoaders.size() < 2) {
+                                contextClassLoaders.add(
+                                        Thread.currentThread().getContextClassLoader());
+                            } else {
+                                throw new RuntimeException("cancel task");
+                            }
+                        },
+                        0,
+                        1,
+                        TimeUnit.MILLISECONDS);
+
+        assertThat(contextClassLoaders)
+                .allSatisfy(
+                        classLoader -> assertThat(classLoader).isSameAs(pretendFlinkClassLoader));
```

Comment:
```suggestion
        final int numberOfScheduledRuns = 2;
        final List<ClassLoader> contextClassLoaders = new ArrayList<>(numberOfScheduledRuns);
        akkaRpcService
                .getScheduledExecutor()
                .scheduleAtFixedRate(
                        () -> {
                            if (contextClassLoaders.size() < numberOfScheduledRuns) {
                                contextClassLoaders.add(
                                        Thread.currentThread().getContextClassLoader());
                            } else {
                                throw new RuntimeException("cancel task");
                            }
                        },
                        0,
                        1,
                        TimeUnit.MILLISECONDS);

        assertThat(contextClassLoaders)
                .hasSize(numberOfScheduledRuns)
                .allSatisfy(
                        classLoader -> assertThat(classLoader).isSameAs(pretendFlinkClassLoader));
```
nit: Adding a final variable makes the test more maintainable (and helps understanding why the 2 is used "everywhere"; although, I'd say it's kind of obvious).
The `hasSize` is not really necessary in my proposal, I guess, because it actually tests the scheduling functionality of the internally used `ScheduledExecutor`.